src/include/simeng/arch/Architecture.hh

```diff
@@ -39,6 +39,8 @@ struct ExceptionResult {
   uint64_t instructionAddress;
   /** Any changes to apply to the process state. */
   ProcessStateChange stateChange;
+  /** The uop to forward onto dependant instructions. */
+  std::shared_ptr<Instruction> uop;
```

Comment:
This is a good point. This was an issue I was experiencing before I implemented this functionality. I will look into it and get back to you

Reply:
After removing and re-testing I remember why this was added. It is true that no other instruction depends on this one due to the flush performed when the exception has been handled. 

However, for the new SME context switches `SMSTART` and `SMSTOP` an exception needs to be raised, and the value in the SVCR system register needs to be updated. Whilst the actual updating of the value occurs in the state change associated with the exception, in the out-of-order core the system register has indicated as "not ready" on the scoreboard. 

As the flush is for all instructions after the current one, without the `uop` being sent to the `forwardOperands` function it appears as though the SVCR register is constantly in flight. In essence, we are releasing the SVCR system register from the scoreboard via the `forwardOperands` function, rather than actually forwarding the result to any dependant instructions.

Reply:
Then I'd argue you should be using `setRegisterReady` rather than the operand forwarding

Reply:
I had overlooked this function completely. Will change it now, thanks

Reply:
Changed. Please see changes in outoforder and inorder core